nixos/tests/systemd-networkd-dhcpserver.nix

```diff
@@ -0,0 +1,61 @@
+# This test predominantly tests systemd-networkd DHCP server, by
+# setting up a DHCP server and client, and ensuring they are mutually
+# reachable via the DHCP allocated address.
+import ./make-test-python.nix ({pkgs, ...}: {
+  name = "systemd-networkd-dhcpserver";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ tomfitzhenry ];
+  };
+  nodes = {
+    router = { config, pkgs, ... }: {
+      virtualisation.vlans = [ 1 ];
+      systemd.services.systemd-networkd.environment.SYSTEMD_LOG_LEVEL = "debug";
+      networking = {
+        useNetworkd = true;
+        useDHCP = false;
+        firewall.enable = false;
+      };
+      systemd.network = {
+        enable = true;
+        networks = {
+          # systemd-networkd will load the first network unit file
+          # that matches, ordered lexiographically by filename.
+          # /etc/systemd/network/{40-eth1,99-main}.network already
+          # exists. This network unit must be loaded for the test,
+          # however, hence why this network is named such.
+          "01-eth1" = {
+            name = "eth1";
+            networkConfig = {
+              DHCPServer = true;
+              Address = "10.0.0.1/24";
+            };
+            dhcpServerConfig = {
+              PoolOffset = 100;
+              PoolSize = 1;
+              # Workaround https://github.com/NixOS/nixpkgs/issues/73504
+              EmitTimezone = false;
```

Comment:
Do we want to wait for upstream to fix this in https://github.com/systemd/systemd/pull/14010#issuecomment-554738488 ? I'm afraid we might forget removing that line once it's fixed upstream…

Reply:
I'm happy to wait for https://github.com/NixOS/nixpkgs/issues/73504 to be resolved.

Reply:
Marking as resolved, since https://github.com/NixOS/nixpkgs/issues/73504 has been resolved.